core/src/main/scala/cats/Composed.scala

```diff
@@ -21,6 +21,8 @@
 
 package cats
 
+import cats.data.INothing
```

Comment:
Is this still a thing? Seems in modern compilers `Nothing` may be working ok.

Reply:
I'm not honestly sure - originally I think this diff used normal `Nothing` and we switched to `INothing` for inference reasons. It may be simpler now than it was at that time.

Reply:
Context:
- https://github.com/typelevel/fs2/pull/2870

Reply:
Hmm I'm giving this a shot but I'm not sure it's possible for this changeset since a few of the instances then return `F[Nothing]` into invariant sites like `Tuple2K`. I'll keep playing with it but it might be a bit finicky

Reply:
Ah nevermind, just need quite a bit more type annotations. Will proceed